pkg/webui/console/components/webhook-form/index.js

```diff
@@ -363,13 +480,46 @@ export default class WebhookForm extends Component {
             sensitive
             code
           />
+          <Form.Field
+            autoFocus
+            title={m.requestBasicAuth}
+            name="headers"
+            label={m.basicAuthCheckbox}
+            onChange={this.handleRequestAuthenticationChange}
+            decode={mapBasicAuthHeaderToBoolean}
+            encode={mapBooleanToBasicAuthHeader}
+            component={Checkbox}
+            tooltipId={tooltipIds.BASIC_AUTH}
+          />
+          {this.state.shouldShowCredentialsInput && (
+            <Form.FieldContainer horizontal>
+              <Form.Field
+                required
+                title={sharedMessages.username}
+                name="headers"
+                decode={mapBasicAuthHeaderToUsername}
+                encode={makeBasicAuthDecoder(true)}
```

Comment:
You should define that outside of the component. Otherwise, the function will be recreated for every rerender.
See my comments above.